scio-test/src/it/scala/com.spotify.scio.values/DistCacheTest.scala

```diff
@@ -0,0 +1,51 @@
+package com.spotify.scio.values
+
+import java.nio.ByteBuffer
+
+import com.google.cloud.dataflow.sdk.util.GcsUtil.GcsUtilFactory
+import com.google.cloud.dataflow.sdk.util.MimeTypes
+import com.google.cloud.dataflow.sdk.util.gcsfs.GcsPath
+import com.spotify.scio.{ScioContext, ScioResult}
+import com.spotify.scio.testing.{ITPipelineSpec, PipelineSpec}
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+class DistCacheTest extends PipelineSpec {
+
+  val distCacheUri = "gs://data-integration-test-us/scio/name-distcache"
+  val cacheData = Seq("name1", "name2")
+
+  "GCS DistCache" should "work" in {
+    runWithDistCache(cacheData, distCacheUri) { sc =>
+      val names = sc.distCache(distCacheUri) { f =>
```

Comment:
move this to `runWithDistCache` and add a param to `fn` like `fn: (ScioContext, DistCache[String]) => T`?